.ci/jenkins/lib/test-matrix.yaml

```diff
@@ -65,50 +66,62 @@ steps:
   - name: Build GPU Test Environment
     parallel: false
     run: |
+      set -x
+      # Clean any leftover nixl-ci-test containers
+      docker ps -q --filter "name=nixl-ci-test" | xargs -r docker rm -f
+      mkdir -p "${WS}"
+      cp -a ${WORKSPACE}/. "${WS}/"
```

Comment:
looks weird, why?

Reply:
`ci-demo` extracts source files to the shared ${WORKSPACE}.
Copy to unique workspace folder per build creates isolation, so containers don't conflict with each other or with workspace cleanup operations.